src/Middlewares/Validator.ts

```diff
@@ -0,0 +1,19 @@
+import { Request, Response, NextFunction } from 'express';
+
+export const validator = (req: Request, res: Response, next: NextFunction) => {
+  const nombre1 = Number(req.query.nombre1);
```

Comment:
Since you define the controller as a class, consider setting the middleware as a class and using the same model.